src/omp/poisson_fft.f90

```diff
@@ -0,0 +1,58 @@
+module m_omp_poisson_fft
+   use m_allocator, only: field_t
+   use m_common, only: dp
+   use m_poisson_fft, only: poisson_fft_t
+   use m_tdsops, only: dirps_t
+
+   use m_omp_common, only: SZ
+
+   implicit none
+
```

Comment:
As for CUDA module, both have an `init` function, it might be worth making this `private` and explicitly exporting subroutines/functions/etc

Reply:
I thought using the `only` keyword strictly when `use`ing a module and explicitly listing things we want to import is safe, but I see what you mean with the `init` function.

Would it work if we declare `private :: init` and continue using `only` whenever we import stuff? With this we can avoid explicitly stating the public ones.

Reply:
I'm not sure, I think you are probably right about using `only`. Making things explicitly `private` as you suggest would also work I think, it depends whether you want the module to be self-contained (default `private`) or only a few things should be inaccessible (default `public`)